test/built-ins/String/prototype/padEnd/observable-operations.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2016 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-string.prototype.padend
+description: String#padEnd should perform observable operations in the correct order
+author: Jordan Harband
+---*/
+
+var log = "";
+
+function createPrimitiveObserver(name, string, value) {
+  return {
+    toString: function () {
+      log += '|toString:' + name;
+      return string;
+    },
+    valueOf: function () {
+      log += '|valueOf:' + name;
+      return value;
+    }
+  };
+};
+
+var receiver = createPrimitiveObserver('receiver', createPrimitiveObserver('toPrimitiveABC', {}, 'abc'));
+
+var fillString = createPrimitiveObserver('fillString', createPrimitiveObserver('toPrimitiveDEF', {}, 'def'));
+
+var maxLength = createPrimitiveObserver(
+    'maxLength',
+    createPrimitiveObserver('toPrimitiveMaxLengthToString', {}, {}),
+    createPrimitiveObserver('toPrimitiveMaxLengthValueOf', {}, 11)
+);
+
+var result = String.prototype.padEnd.call(receiver, fillString, maxLength);
+
+assert.sameValue(result, 'defdefdeabc');
```

Comment:
hmm… the chain is such that `receiver` should ToPrimitive to `'abc'`, and `fillString` to `'def'`